distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -1049,15 +1048,16 @@ type Column
          supported type is chosen and a `Inexact_Type_Coercion` problem is
          reported.
     cast : Value_Type -> Problem_Behavior -> Column ! Illegal_Argument | Inexact_Type_Coercion | Lossy_Conversion
-    cast self value_type=self.value_type on_problems=Problem_Behavior.Report_Warning =
-        dialect = self.connection.dialect
-        type_mapping = dialect.get_type_mapping
-        target_sql_type = type_mapping.value_type_to_sql value_type on_problems
-        target_sql_type.if_not_error <|
-            infer_from_database new_expression =
-                SQL_Type_Reference.new self.connection self.context new_expression
-            new_column = dialect.make_cast self.as_internal target_sql_type infer_from_database
-            Column.Value new_column.name self.connection new_column.sql_type_reference new_column.expression self.context
+    cast self value_type=Value_Type.Char on_problems=Problem_Behavior.Report_Warning =
```

Comment:
Hmm, but why?

We wanted to have as many defaults on nodes as possible, right? So that they preview nicely in the CB when 'pre-selected'.

While for `parse` we should remove the default in Database, because there is no good choice; I think for `cast` `Char` is a pretty good choice - converting _to Text_ should work in baiscally every case and is a good example of what cast does.

I can ofc change it if you think this is the right approach, but IMO this particular default is OK, so I'd like to confirm that you really want this and would appreciate understanding why.

Reply:
I feel like in `cast` it is a real choice of the user so I felt like it should be a users explicit choice.

Reply:
I see. But if we go by this heuristic, shouldn't we also remove defaults from stuff like `select_columns`, `remove_columns` and many more? There it is even more of an explicit user's choice which columns they want to remove - defaulting to the first one is just an arbitrary default we have to show the functionality working in the preview.

_(I don't want to be stubborn but I want us to be consistent and really understand the reasoning for why we want to include defaults in X but not in Y.)_

Reply:
I agree - and ok lets go with Text as the default I guess as its harmless.

I'm not sure we should have defaults on everything and this feels a more aggressive default.

Reply:
> I'm not sure we should have defaults on everything 

Yeah, I think I agree.


> and this feels a more aggressive default.

But I cannot understand this part - hence my opposition.

It's not super important, and I guess a bit subjective so it may be hard to explain - but could you try to narrow down what makes this a 'more aggresive' default?

I think my dissonance comes from the fact that IMO `select_columns` and `remove_columns` defaults are even 'more aggresive' than this one. Or maybe you agree that these are 'aggressive defaults' too?

Reply:
(I know it's not the most important discussion, but just trying to understand your perspective so I can understand the decisions better)

Reply:
It feels like it's changing data in a way - in select it's basically a no-op (we just pick a column from the list).
On a billion-row table, `cast` is a chunky complex operation and hence I guess felt like worth getting user to choose what they want to do. Unlike `parse`, we can't guess.

Reply:
Ah, I didn't think of the _operation cost_! That is a very good point indeed - thanks for explaining it.

So I can indeed remove the default, we indeed don't want to wait for a very long operation to complete just to switch the type...